### DISCUSSION TITLE
Keepass / CSV Imports & Exports

### DISCUSSION THREAD

**Discussion by MisterPlop** [nilsteampassnet/TeamPass#3530] [Q&A]
Hi there,
(not so) Quick question about imports & exports, on TeamPass 3.0.0.22

I've been meddling around and couldn't get to properly use these functionalities

**Keepass:**
When I'm importing a keepass.xml file with, inside, folders & subfolders, folders are created, but no subfolder is created. 
Even so, after creating them manually and assigning weak password policy then re-importing, this just creates duplicate empty folders  and leave existing ones aside (with the option "folders have duplicate names toggled" off).
Tried looking at permissions of auto-created folders but they look good to me, similar to the parent folder.
Though limiting myself to folders works well so it's not that much of an issue; i'll be able to import and move them in TeamPass subfolders later.

**CSV:**
When importing a .csv file exported from another TeamPass instance, ID becomes Label, Label becomes description etc. 
The table from the TeamPass' exported .csv is as such : 
> id,label,description,pw,login,restricted_to,perso,url,email,kb,tag,folder

But in the import.js.php file it seems we lose folder hierarchy, and instead ask for : 
(id is used but not pushed, which makes sense)
>  label: $(this).data('label'),
>  login: $(this).data('login'),
>  pwd: $(this).data('pwd'),
>  url: $(this).data('url'),
>  comment: $(this).data('comment'),
>  row: $(this).data('row'),

Am I doing something wrong? Should I open a new issue?
Thank you !

**Comment by nilsteampassnet** [comment]
Hello @MisterPlop 

Regarding KeePass, I need to perform new tests to check this point.

Regarding CSV feature, the usecase behind "import" feature was not to reimport a Teampass exported CSV file.
So indeed when performing an extract, the first column is the item ID.
But when importing, the format expected does not require this ID column.
![image](https://user-images.githubusercontent.com/1197546/224476901-2742ecd3-86ca-4d70-b7f0-aa3617dd6a1b.png)

I hope this is more clear and also I will add this in new documentation

**Reply by Francky017** [reply]
Hi @nilsteampassnet,

I'm facing to a almost similar issue.

When I try to import by CSV using the CSV structure in your message, I'm not able to create automaticaly all folder and subfolders.

It seems in this "label,login,password,url,comments" we don't have possibility to create folder and subfolders.

Thanks for your answer in advance.
